modules/freetype/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+set(the_description "FreeType module. It enables to draw strings with outlines and mono-bitmaps/gray-bitmaps.")
+
+if(PKG_CONFIG_FOUND)
+  pkg_search_module(FREETYPE freetype2)
+  pkg_search_module(HARFBUZZ harfbuzz)
+endif()
+
+if(NOT FREETYPE_FOUND)
+  message(STATUS "freetype2:   NO")
+else()
+  message(STATUS "freetype2:   YES")
+endif()
+
+if(NOT HARFBUZZ_FOUND)
+  message(STATUS "harfbuzz:    NO")
+else()
+  message(STATUS "harfbuzz:    YES")
+endif()
+
+
+if( FREETYPE_FOUND AND HARFBUZZ_FOUND )
+  include_directories(${FREETYPE_INCLUDE_DIRS} ${HARFBUZZ_INCLUDE_DIRS} )
```

Comment:
The right order is:
```
ocv_define_module(...)
ocv_include_directories( ... )
```

Could you try to move these includes from `precomp.hpp` into implementation .cpp file?